src/fdk.ts

```diff
@@ -226,10 +227,15 @@ Command.prototype.asyncAction = async function (asyncFn: Action) {
 
 //   return [''].join() + '\n';
 // };
-
 export async function init(programName: string) {
-    await checkForLatestVersion();
-    
+    await Promise.race([
+        checkForLatestVersion(),
+        new Promise((resolve, reject) => setTimeout(() => {
```

Comment:
Create new utility function for this kind of use case which can be used by others as well @bhargavprajapatiFynd

Reply:
Here, we can't create a separate function as there is no way to stop an already running function once it has started executing.

so I have declared a separate variable (version_check_timed_out) which is used to stop function execution if there is a timeout.

Reply:
Moved race condition logic inside version check function